tests/rptest/services/redpanda.py

```diff
@@ -306,21 +314,23 @@ def load_context(self, logger, test_context):
             self.GLOBAL_S3_REGION_KEY, None)
 
         # Enable S3 if AWS creds were given at globals
-        if cloud_storage_access_key and cloud_storage_secret_key:
+        if dedicated_nodes:
```

Comment:
using `dedicated_nodes` global ducktape config setting because with this PR there are now 2 ways to connect to s3: 1. access key + secret, 2. iam roles. since `dedicated_nodes` will have `true` value in either of those situations, this seemed like the way to proceed.